torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -274,9 +281,19 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::send(
   auto requestId = nextId();
   auto future = std::make_shared<FutureMessage>();
   if (message.isRequest()) {
+    // millisecond level precision of when request started.
+    auto futureStartTime =
```

Comment:
Would it be more convenient to lookup later, if they are keyed by "end time"? I am considering RPCs with overwriting timeouts.

Reply:
When we add support for detecting and marking timed out RPCs, we'd like them to be in the `map` by start time, so that it is faster to look up the RPC(s) which started earliest. Then if we determine that RPC has not timed out, we can save ourselves the linear search, which is beneficial since we can have a large number of RPCs.